tests/internal/services/sso_service_test.go

```
@@ -102,7 +101,113 @@ func TestGetAllUsersWithExistingUsers(t *testing.T) {
+			assert.Equal(t, tc.expected, actual, "\n%s - actual: %v, expected: %v", tc.name, actual, tc.expected)
+		})
+	}
+}
+
+func TestGetUserByIDNotFound(t *testing.T) {
+	const testUserID = 1
+	ssoService := &services.CommonSsoService{
+		SsoRepository: &mocks.MockedSsoRepository{
+			UsersStorage: map[int]*ssoentities.User{},
+		},
+	}
+
+	userID, err := ssoService.GetUserByID(testUserID)
+	assert.Nil(t, userID, "should return nil for user with ID=%d", testUserID)
+	assert.IsType(t, &customerrors.UserNotFoundError{}, err)
+	assert.Equal(t, "user not found", err.Error())
+}
+
+func TestLoginUser(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    ssoentities.LoginUserDTO
+		expected string
+		wantErr  error
+	}{
+		{
+			name:     "should return token",
+			input:    ssoentities.LoginUserDTO{Email: "test@example.com", Password: "password"},
+			expected: "someToken",
+			wantErr:  nil,
+		},
+		{
+			name:     "should return error if user not found",
+			input:    ssoentities.LoginUserDTO{Email: "nonexistent@example.com", Password: "password"},
+			expected: "",
+			wantErr:  &customerrors.UserNotFoundError{},
+		},
+		{
+			name:     "should return error if password incorrect",
+			input:    ssoentities.LoginUserDTO{Email: "test@example.com", Password: "wrongPassword"},
+			expected: "",
+			wantErr:  &customerrors.InvalidPasswordError{},
+		},
+	}
+
+	ssoRepository := &mocks.MockedSsoRepository{
+		UsersStorage: map[int]*ssoentities.User{
+			1: {
+				Email:    "test@example.com",
+				Password: "password",
+			},
+		},
+	}
+
+	ssoService := &services.CommonSsoService{SsoRepository: ssoRepository}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := ssoService.LoginUser(tc.input)
+			if tc.wantErr != nil {
+				require.Error(t, err, "Should return an error")
+				assert.Equal(t, tc.wantErr, err)
+			} else {
+				require.NoError(t, err, "Should return no error")
+				assert.Equal(
```

вот это можно вынести за передел условнйо конструкции. Проверятсья должно в обоих случаях
			if tc.wantErr != nil {
				require.Error(t, err, "Should return an error")
				assert.Equal(t, tc.wantErr, err)
			} else {
				require.NoError(t, err, "Should return no error")
			}
			
			assert.Equal(
					t,
					tc.expected,
					actual,
					"\n%s - actual: %v, expected: %v", tc.name, actual, tc.expected)

---

Исправил.